src/test/soundproxy_test.cpp

```diff
@@ -187,10 +190,13 @@ TEST_F(SoundSourceProxyTest, open) {
                 // skip test file
                 continue;
             }
+            couldOpenFile = true;
```

Comment:
MAD will open the file successfully and FFmpeg is skipped.

Unfortunately, the lazy test behavior was needed for the AAC/ALAC files that both share the same `.m4a` file extension.

Reply:
> MAD will open the file successfully and FFmpeg is skipped. No errors, unless you disable all other MP3 decoders that have a higher priority than FFmpeg.

Yes, that was the intention here. It's not super important what decoder is used for what format, its only important that we can open the file with at least one. The other tests are care about the actual encoder and what offset it introduces/requires.

> Although I personally consider it as the must reliable decoder for MP3, MP4, and ALAC (at least on Linux/Windows).

I agree, thats why I compile with ffmpeg and without MAD.

> Unfortunately, the lazy test behavior was needed for the AAC/ALAC files that both share the same .m4a file extension.

What do you mean with "lazy test behavior"? The fallback approach?

Reply:
Most tests don't fail if opening fails. You don't notice if a file that could previously be opened can no longer be decoded. Otherwise you would need to explicitly which combinations are expected to fail and should be skipped. This would be preferable to detect regressions.